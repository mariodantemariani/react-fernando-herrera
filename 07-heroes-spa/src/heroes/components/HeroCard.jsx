import PropTypes from "prop-types";
import { Link } from "react-router-dom";
import "animate.css";

const CharactersByHero = ({ characters, alter_ego }) => {
  return characters === alter_ego ? (
    <></>
  ) : (
    <p className="card-text">{characters}</p>
  );
};

export const HeroCard = ({
  id,
  superhero,
  alter_ego,
  first_appearance,
  characters,
}) => {
  const heroImageUrl = `/assets/heroes/${id}.jpg`;
  const heroUrl = `/hero/${id}`;

  ///const charactersByHero = <p className="card-text">{characters}</p>;

  return (
    <div className="col animate__animated animate__fadeIn">
      <div className="card ms-3" style={{ maxWidth: 540 }}>
        <div className="row no-gutters">
          <div className="col-md-4">
            <img src={heroImageUrl} className="card-img" alt={superhero} />
          </div>
          <div className="col-md-8">
            <div className="card-body">
              <h5 className="card-title">{superhero}</h5>
              <p className="card-text">{alter_ego}</p>
              {/* {alter_ego !== characters && charactersByHero} */}
              <CharactersByHero characters={characters} alter_ego={alter_ego} />
              <p className="card-text">
                <small className="text-muted">{first_appearance}</small>
              </p>
              <Link to={heroUrl}>Más...</Link>
              {/* <button className="btn btn-outline-primary mt-5" >Ver más...</button> */}
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

CharactersByHero.propTypes = {
  characters: PropTypes.string.isRequired,
  alter_ego: PropTypes.string.isRequired,
};

HeroCard.propTypes = {
  id: PropTypes.string.isRequired,
  superhero: PropTypes.string.isRequired,
  alter_ego: PropTypes.string.isRequired,
  first_appearance: PropTypes.string.isRequired,
  characters: PropTypes.string.isRequired,
  publisher: PropTypes.string.isRequired,
};
