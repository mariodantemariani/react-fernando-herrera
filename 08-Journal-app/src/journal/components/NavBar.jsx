import { AppBar, Grid, IconButton, Toolbar, Typography } from "@mui/material";
import { LogoutOutlined, MenuOutlined } from "@mui/icons-material";

export const NavBar = ({ drawerWidth }) => {
  return (
    <AppBar
      position="fixed"
      sx={{
        width: { sm: `calc(100% - ${drawerWidth}px)` },
        ml: { sm: `${drawerWidth}px` },
      }}
    >
      <Toolbar>
        <IconButton
          color="inherit"
          aria-label="menu"
          edge="start"
          sx={{ mr: 2, display: { sm: "none" } }}
        >
          <MenuOutlined />
        </IconButton>
        <Grid
          container
          direction="row"
          justifyContent="space-between"
          alignItems="center"
        >
          <Typography variant="h6" noWrap component="div" sx={{ flexGrow: 1 }}>
            JournalApp
          </Typography>
          <IconButton color="error">
            <LogoutOutlined />
          </IconButton>
        </Grid>
      </Toolbar>
    </AppBar>
  );
};
