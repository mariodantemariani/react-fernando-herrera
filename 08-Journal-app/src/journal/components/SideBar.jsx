import { useDispatch, useSelector } from "react-redux";

import {
  Box,
  Divider,
  Drawer,
  Grid,
  List,
  ListItem,
  ListItemButton,
  ListItemIcon,
  ListItemText,
  Toolbar,
  Typography,
} from "@mui/material";

import { SideBarItem } from "./SideBarItem";

export const SideBar = ({ drawerWidth = 240 }) => {
  const { displayName } = useSelector((state) => state.auth);
  const { notes } = useSelector((state) => state.journal);

  return (
    <Box
      component={"nav"}
      sx={{ width: { sm: drawerWidth }, flexShrink: { sm: 0 } }}
    >
      <Drawer
        variant="permanent"
        open
        sx={{
          display: { xs: "block" },
          "& .MuiDrawer-paper": { boxSizing: "border-box", width: drawerWidth },
        }}
      >
        <Toolbar>
          <Typography variant="h6" noWrap component="div">
            {displayName}
          </Typography>
        </Toolbar>
        <Divider />

        <List>
          {notes.map((note) => (
            <SideBarItem key={note.id} {...note} />
            // <ListItem key={note.id} disablePadding>
            //   <ListItemButton>
            //     <ListItemIcon>
            //       <TurnedInNot />
            //     </ListItemIcon>
            //     <Grid container>
            //       <ListItemText primary={note.title} />
            //       <ListItemText
            //         secondary={"Exercitation cillum irure elit consectetur."}
            //       />
            //     </Grid>
            //   </ListItemButton>
            // </ListItem>
          ))}
        </List>
      </Drawer>
    </Box>
  );
};
