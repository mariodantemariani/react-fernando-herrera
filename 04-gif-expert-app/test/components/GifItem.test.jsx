import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe("Pruebas en GifItem", () => {
  const title = "test";
  const url = "http://localhost/test"; //no sé que me añade http://localhost/ y no a Fernando
  const id = "test";

  test("Should match with snapshoot", () => {
    const { container } = render(<GifItem title={title} url={url} id={id} />);

    //asserts
    expect(container).toMatchSnapshot();
  });

  test("should show the image and ALT correctly", () => {
    render(<GifItem title={title} url={url} />);
    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test("should show the title", () => {
    render(<GifItem title={title} url={url} id={id} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
