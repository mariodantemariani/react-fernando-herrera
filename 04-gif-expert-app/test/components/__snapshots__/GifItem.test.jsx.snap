// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItem Should match with snapshoot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="test"
      src="http://localhost/test"
    />
    <p>
      test
    </p>
  </div>
</div>
`;
